### PR TITLE
Add docs about Docker stemcells

### PR DESCRIPTION
**Pull request by viovanov**

### CODE REVIEW

File: docs/stemcells.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Prior to version `5.0.0`, fissile would build its own compilation and stemcell
+layers, using hardcoded scripts, that only worked for Ubuntu Trusty.
+
+Starting with version `5.0.0`, fissile no longer builds these images. The user
+is responsible with providing an image that contains the same packages and
+dependencies as a regular [BOSH stemcell](https://github.com/cloudfoundry/bosh-linux-stemcell-builder/tree/master/stemcell_builder/stages), plus:
+- prerequisites: `libopenssl-devel`, `gettext-tools`
+- [`configgin`](https://rubygems.org/gems/configgin) - a ruby gem that processes
+  BOSH erb templates (this means ruby and bundler need to be there as well)
+- [`dumb-init`](https://github.com/Yelp/dumb-init) - a tool used as PID 1 for managing processes in a container
```
Comment (aduffeck) [inline docs/stemcells.md]: We're currently calling `dumb-init` from a [specific path](https://github.com/SUSE/fissile/blob/22080c226448baf123e4a4920bffe53d3f2b4088/scripts/dockerfiles/Dockerfile-role#L12). Should we mention this here?

File: docs/stemcells.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Implementations
+
+- [OpenSUSE](https://github.com/SUSE/fissile-stemcell-openSUSE/blob/42.2/Dockerfile)
+- [Ubuntu](https://github.com/cloudfoundry-community/fissile-stemcell-ubuntu)
+
+Both of these are built from from the actual BOSH stemcells, you can find the
```
Comment (aduffeck) [inline docs/stemcells.md]: `from from` -> `from`

> from the actual BOSH stemcells

To be precise it is built from one of the intermediate blobs that are generated when building the actual stemcells, not the stemcell itself. It is said in the next sentence, but at this point it's not considered a stemcell yet, at least in the upstream terminology.

File: docs/stemcells.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Implementations
+
+- [OpenSUSE](https://github.com/SUSE/fissile-stemcell-openSUSE/blob/42.2/Dockerfile)
+- [Ubuntu](https://github.com/cloudfoundry-community/fissile-stemcell-ubuntu)
+
+Both of these are built during the same process that builds the actual BOSH
+stemcells; you can find the pipeline for these [here](https://ci.from-the.cloud/teams/main/pipelines/bosh-os-images).
+The CPI specific dependencies are not required for Docker Stemcells, so we use
+the BOSH stemcells before they are differentiated for each supported IaaS.
```
Comment (dragonchaser) [inline docs/stemcells.md]: Should we mention the splatform repository on dockerhub as stemcell source?

Comment (viovanov) [inline docs/stemcells.md]: @dragonchaser I don't think we need to - those will be mentioned/used in the SCF repo

Comment (dragonchaser) [inline docs/stemcells.md]: Ok, then this lgtm

**Pull request merged**